From 9cf97b7b4b7bc14ceea6bb642b822f888293d555 Mon Sep 17 00:00:00 2001
From: Megha Dey <megha.dey@linux.intel.com>
Date: Tue, 19 Apr 2016 12:19:51 -0700
Subject: [PATCH 2/2] ndctl: check dependencies for docs only if needed

asciidoc and xmlto are needed to build the documentation for ndctl.
However, we should not depend on them if we are not building the
documentation. Thus, remove checks for them if --disable-docs is used.

The asciidoc package is fairly large. LUV is an automated tool to run
tests and we dont have a command line to do a man ndctl. Thus, we
can safely disable the documentation to be included in the ndctl package.
Users can get the necessary information for the ndctl homepage.

Signed-off-by: Megha Dey <megha.dey@linux.intel.com>
---
 configure.ac | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1ab3768..b7de62d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -32,14 +32,18 @@ AC_PREFIX_DEFAULT([/usr])
 
 AC_PROG_SED
 AC_PROG_MKDIR_P
-AC_CHECK_PROG(ASCIIDOC, [asciidoc], [$(which asciidoc)], [missing])
-if test "x$ASCIIDOC" = xmissing; then
-       AC_MSG_ERROR([asciidoc needed to build documentation])
-fi
-AC_SUBST([ASCIIDOC])
-AC_CHECK_PROG(XMLTO, [xmlto], [$(which xmlto)], [missing])
-if test "x$XMLTO" = xmissing; then
-       AC_MSG_ERROR([xmlto needed to build documentation])
+
+if test "x$enable_docs" = "xyes"; then
+	AC_CHECK_PROG(ASCIIDOC, [asciidoc], [$(which asciidoc)],[missing])
+	if test "x$ASCIIDOC" = xmissing; then
+		AC_MSG_ERROR([asciidoc needed to build documentation])
+	fi
+	AC_SUBST([ASCIIDOC])
+	AC_CHECK_PROG(XMLTO, [xmlto], [$(which xmlto)], [missing])
+	if test "x$XMLTO" = xmissing; then
+		AC_MSG_ERROR([xmlto needed to build documentation])
+	fi
+	AC_SUBST([XMLTO])
 fi
 AC_SUBST([XMLTO])
 
-- 
1.9.1

